### ISSUE TITLE
Provide the timelimit as a variable for the mesage

### ISSUE THREAD

**Issue by FrTerstappen** [ironpeak/slow_function_warning#5]
I really like the idea of this crate.

Would it be possible to provide the time limit as a variable to use in the message?

What I imagine is changing the generated code from

```rust
[...]
if start.elapsed().as_nanos() > 1000000 {
    let module = "module name";
[...]
```

to

```rust
[...]
let limit_nano_seconds = 1000000;
if start.elapsed().as_nanos() > limit_nano_seconds {
    let module = "module name";
[...]
```

(Or even better all forms like for the elapsed time)

What I want to achieve is a message like: `System test::test ran for 2ms (limit: 1ms)`

**Comment by ironpeak** [comment]
I like the idea, I've started work on it here if you want to try it out:
https://github.com/ironpeak/slow_function_warning/pull/6/files

I haven't had time to test and document it yet.

**Comment by ironpeak** [comment]
Published an RC with the changes: `0.6.0-rc.1`

**Comment by FrTerstappen** [comment]
Thank you very much for the fast response.

I just tried the RC `0.6.0-rc.1`. This works very good.

It just comes with one small problem. I had to add `use std::time::Duration;` because it is not fully qualified in the generated code unlike the usage of `std::time::Instant`.
This should be a very easy fix.

**Comment by ironpeak** [comment]
Good catch!

I created a new RC `0.6.0-rc.2` where everything should be fully qualified (added tests for it as well).

**Comment by FrTerstappen** [comment]
I just tested your latest changes. Everything looks good to me.

**Comment by ironpeak** [comment]
Released version `0.6.0`
